frost/signer.go

```diff
@@ -430,21 +495,48 @@ func (s *Signer) deriveInterpolatingValue(xi uint64, L []uint64) (*big.Int, erro
 		den.Mod(den, order)
 	}
 
-	// if x_i not in L:
-	//     raise "invalid parameters"
-	if !found {
-		return nil, fmt.Errorf(
-			"invalid parameters: xi=[%v] not present in L=[%v]",
-			xi,
-			L,
-		)
-	}
-
 	// value = numerator / denominator
 	denInv := new(big.Int).ModInverse(den, order)
 	res := new(big.Int).Mul(num, denInv)
 	res = res.Mod(res, order)
 
 	// return value
-	return res, nil
+	return res
+}
+
+// computeChallenge implements def compute_group_commitment(commitment_list,
+// binding_factor_list) from [FROST] as defined in section 4.6. Signature
+// Challenge Computation.
+func (s *Signer) computeChallenge(
+	message []byte,
+	groupCommitment *Point,
+) *big.Int {
+
+	// From [FROST]:
+	//
+	// 4.6.  Signature Challenge Computation
+	//
+	//   This section describes the subroutine for creating the per-message
+	//   challenge.
+	//
+	//   Inputs:
+	//     - group_commitment, the group commitment, an Element.
+	//     - group_public_key, the public key corresponding to the group signing
+	//       key, an Element.
+	//     - msg, the message to be signed, a byte string.
+	//
+	//   Outputs:
+	//     - challenge, a Scalar.
+	//
+	// def compute_group_commitment(commitment_list, binding_factor_list)
+
+	curve := s.ciphersuite.Curve()
+	// group_comm_enc = G.SerializeElement(group_commitment)
+	groupCommitmentEncoded := curve.SerializePoint(groupCommitment)
+	// group_public_key_enc = G.SerializeElement(group_public_key)
+	publicKeyEncoded := curve.SerializePoint(s.publicKey)
+	// challenge_input = group_comm_enc || group_public_key_enc || msg
+	// challenge = H2(challenge_input)
+	// return challenge
+	return s.ciphersuite.H2(groupCommitmentEncoded, publicKeyEncoded, message)
```

Comment:
It is possible that some suites may use something weird for `H2` which would be incompatible with this, but I think for the most part this is likely to be fine. Something we may want to consider is whether `H2` should take in `(Point, Point, []byte)` arguments instead, but it shouldn't be a blocker.